src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -185,11 +185,21 @@ private SyntaxNode MutateForStatement(ForStatementSyntax forStatement)
 
             var originalFor = forWithMutantIncrementors.GetCurrentNode(forStatement);
             // now we generate a mutant for the remainder of the for statement
-            var mutatedFor = forStatement.TrackNodes(forStatement.Condition, forStatement.Statement);
-            mutatedFor = mutatedFor.ReplaceNode(mutatedFor.GetCurrentNode(forStatement.Condition),
-                Mutate(forStatement.Condition));
-            mutatedFor = mutatedFor.ReplaceNode(mutatedFor.GetCurrentNode(forStatement.Statement),
-                Mutate(forStatement.Statement));
+
+            ForStatementSyntax mutatedFor;
+            if (forStatement.Condition == null)
+            {
+                mutatedFor = forStatement.ReplaceNode(forStatement.Statement,
```

Comment:
I will refactor this bit. Not sure if this what you have in mind. please comment

Reply:
```cs
mutatedFor = forStatement.TrackNodes(forStatement.Condition, forStatement.Statement);
if (forStatement.Condition != null) {
    mutatedFor = mutatedFor.ReplaceNode(mutatedFor.GetCurrentNode(forStatement.Condition), Mutate(forStatement.Condition));
}
mutatedFor = mutatedFor.ReplaceNode(mutatedFor.GetCurrentNode(forStatement.Statement), Mutate(forStatement.Statement));
```

This would functionally do the same I think?

Reply:
no, you can't track a null node